packages/react-router/src/routeRegistry.ts

```diff
@@ -0,0 +1,24 @@
+import { RouteObject } from "react-router-dom";
+
+export type Route = RouteObject;
+
+export type RootRoute = Route & {
+    hoist?: boolean;
+};
+
+export class RouteRegistry {
+    private _routes: RootRoute[];
+
+    constructor() {
+        this._routes = [];
+    }
+
+    add(routes: RootRoute[]) {
+        // Create a new array so the routes array is immutable.
+        this._routes = [...this._routes, ...routes.filter(x => x)];
```

Comment:
same as navigationItems.filter

...navigationItems.filter(x => x) feels odd. I understand you want to filter ou NavigationItems that are undefined. However, according to the typings, undefined is not allowed in the array. So if we feel like they still might be undefined, i would probably do the validation when first loading the navigation items, not here